### DISCUSSION TITLE
Clearing the input/output window using code?

### DISCUSSION THREAD

**Discussion by rajrao** [pybricks/support#1074] [Q&A] [answered]
Is there a way to clear the input/output window via code? I would like to clear the window whenever I start my program. Currently the window keeps the previous outputs and I have to manually clear the input/output window.

**Comment by dlech** [comment, marked as answer]
Try `print("\x1b[H\x1b[2J", end="")`. These are terminal escape codes to clear the terminal.

https://unix.stackexchange.com/questions/124762/how-does-clear-command-work

**Reply by rajrao** [reply]
Thank you for that answer and the link. There is a typo in your code. I was able to make it work using the following:
```print("\x1b[H\x1b[2J", end="")```
2J instead of 3J. 3J wasnt clearing the screen and may have been clearing just the top 2 lines.

Any chance a clear() or clearConsole() could be added to built-in commands? If so, would it be done here: https://github.com/pybricks/pybricks-api/blob/master/src/ubuiltins/__init__.py (I could submit a pull request for it)

**Reply by dlech** [reply]
Glad to hear you got it working. 

Since the hubs have quite limited flash memory, we can't say yes to every suggestion for new APIs. I would suggest opening a feature request for it at https://github.com/pybricks/support/issues. If there is high enough demand we could consider it.

Since sharing code between files is possible, in the meantime, you could make a `util.py` file with this function in it that can be shared between your projects.

**Reply by rajrao** [reply]
Can you fix the typo, as your reply is marked as the answer.

**Reply by dlech** [reply]
Done.

**Comment by BertLindeman** [comment]
On beta.pybricks.com the esc H esc 3J moves the cursor to the top, but the data remains on screen.
Following text overwrites the previous. 
Used firmware 
```
('primehub', '3.3.0b4', 'v3.3.0b4 on 2023-04-21')
```

Ran the following to test and to me it seems that esc H esc 2J does what you want:
```python
from pybricks.tools import wait 

print("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx")
wait(1000)
print("\x1b[H\x1b[3J", end="")
print("yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy")
wait(1000)
print("\x1b[H\x1b[3J", end="")
print("zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz")
wait(1000)
print("\x1b[H\x1b[1J", end="")
print("Done esc 1J")
wait(1000)
print("\x1b[H\x1b[3J", end="")
print("Done esc 3J")
wait(1000)
print("\x1b[H\x1b[2J", end="")
wait(1000)
print("After esc 2J - Do once more to get an empty output screen")
wait(1000)
print("\x1b[H\x1b[2J", end="")
```

Ran this on code.python.com  on firmware:
```
('cityhub', '3.3.0b3', 'v3.3.0b3-60-gf6379307 on 2023-04-14')
```

Same result.

[EDIT] On windows 11 with Google Chrome Version 113.0.5672.93 (Official Build) (64-bit)

**Comment by bcfletcher** [comment]
I'm using the pybricks 3.6 beta with Chrome on MacOS 15.3.1 and it works but I have to do this:

```python
    print("\x1b[H\x1b[3J", end="")
    print("\x1b[H\x1b[2J", end="")
```
